R/R/metrics.R

```diff
@@ -0,0 +1,141 @@
+#'
+#' @details
+#' Here, as a notation, we assume that \eqn{y} is the actual value and \eqn{yhat} is the forecast value.
+#'
+#' Mean Error (ME, \code{me})
+#'
+#' The Mean Error (ME)  is defined by the formula:
+#' \deqn{ \frac{1}{n} \sum_{t=1}^{n} y_{t}-yhat_{t} .}
+#'
+#' Mean Squared Error (MSE, \code{mse})
+#'
+#' The Mean Squared Error (MSE)  is defined by the formula:
+#' \deqn{ \frac{1}{n} \sum_{t=1}^{n} (y_{t}-yhat_{t})^2 .}
+#'
+#' Root Mean Square Error (RMSE, \code{rmse})
+#'
+#' Root Mean Square Error (RMSE) is define by the formula:
+#' \deqn{ \sqrt{\frac{1}{n} \sum_{t=1}^{n} (y_{t}-yhat_{t})^2} .}
+#'
+#' Mean Absolute Error (MAE, \code{mae})
+#'
+#' The Mean Absolute Error (MAE) is defined by the formula:
+#' \deqn{ \frac{1}{n} \sum_{t=1}^{n} | y_{t}-yhat_{t} | .}
+#'
+#' Mean Percentage Error (MPE, \code{mpe})
+#'
+#' The Mean Percentage Error (MPE) is usually expressed as a percentage
+#' and is defined by the formula:
+#' \deqn{ \frac{100}{n} \sum_{t=1}^{n} \frac {y_{t}-yhat_{t}}{y_{t}} .}
+#'
+#' Mean Absolute Percentage Error (MAPE, \code{mape})
+#'
+#' The Mean absolute Percentage Error (MAPE), also known as Mean Absolute Percentage Deviation (MAPD), is usually expressed as a percentage,
+#' and is defined by the formula:
+#' \deqn{ \frac{100}{n} \sum_{t=1}^{n} | \frac {y_{t}-yhat_{t}}{y_{t}}| .}
+#'
+#' @param fcst Dataframe output of `predict`.
+#'
+#' @return metrics value (numeric)
+#'
+#'@examples
+#'\dontrun{
+#' # Create example model
+#' library(readr)
+#' library(prophet)
+#' df <- read_csv('../tests/testthat/data.csv')
+#' m <- prophet(df)
+#' future <- make_future_dataframe(m, periods = 365)
+#' forecast <- predict(m, future)
+#' all_metrics(forecast)
+#' df.cv <- cross_validation(m, horizon = 100, units = 'days')
+#' all_metrics(df.cv)
+#' # You can check your models's accuracy using me, mse, rmse ...etc.
+#' print(rmse(m))
+#'}
+#' @name metrics
+NULL
+
+#' @rdname metrics
+#' @export
+me <- function(fcst)
+{
+  df <- create_metric_data(fcst)
+  mean(df$y-df$yhat)
+}
+
+#' @rdname metrics
+#' @export
+mse <- function(fcst)
+{
+  df <- create_metric_data(fcst)
+  mean((df$y-df$yhat)^2)
+}
+
+#' @rdname metrics
+#' @export
+rmse <- function(fcst)
+{
+  sqrt(mse(fcst))
+}
+
+#' @rdname metrics
+#' @export
+mae <- function(fcst)
+{
+  df <- create_metric_data(fcst)
+  mean(abs(df$y-df$yhat))
+}
+
+#' @rdname metrics
+#' @export
+mpe <- function(fcst)
+{
+  df <- create_metric_data(fcst)
+  100*mean((df$y-df$yhat)/df$y)
+}
+
+#' @rdname metrics
+#' @export
+mape <- function(fcst)
+{
+  df <- create_metric_data(fcst)
+  100*mean(abs(df$y-df$yhat)/df$y)
+}
+
+#' @rdname metrics
+#' @export
+all_metrics <- function(fcst)
+{
+  # Define all metrics functions as a character
+  metrics <- c("me", "mse", "rmse", "mae", "mpe", "mape")
+  # Convert character to function and evalate each metrics
+  result <- purrr::map(metrics, ~ rlang::eval_tidy(rlang::sym(.x))(fcst))
```

Comment:
`rlang::eval_tidy(rlang::sym(.x))` is more *modern* R programming style compared to `eval(parse(text=.x))`